### PR TITLE
Change dropdown menus to accessible input fields

### PR DESCRIPTION
**Pull request by MateuszMalisz**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [ ] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

Modified dropdown options to provide user with a list which can be cycled through using arrow keys. This improves the experience for TTS.

###### Change Log  

- Change dropdown fields in terminal installer to special input fields.

###### Does this affect the toolchain?  

NO

### CODE REVIEW

File: toolkit/scripts/toolchain.mk [hunk 1]
```diff
@@ -159,6 +159,8 @@ $(final_toolchain): $(raw_toolchain) $(BUILD_SRPMS_DIR)
 	$(if $(filter y,$(UPDATE_TOOLCHAIN_LIST)), ls -1 $(toolchain_build_dir)/built_rpms_all > $(MANIFESTS_DIR)/package/toolchain_$(build_arch).txt)
 	touch $@
 
+.SILENT: $(toolchain_rpms)
```
Comment (MateuszMalisz) [inline toolkit/scripts/toolchain.mk:162]: This disables the verbose mode of the toolchain extraction. Makes the output much smaller and much easier to understand.

Comment (PawelWMS) [reply toolkit/scripts/toolchain.mk:162]: Oh, that's handy. I'm stealing that for some of my logging changes.;)

File: toolkit/tools/imagegen/attendedinstaller/views/diskview/manualpartitionwidget/manualpartitionwidget.go [hunk 2]
```diff
@@ -127,26 +130,13 @@ func (mp *ManualPartitionWidget) Initialize(backButtonText string, sysConfig *co
...
-		SetOptions(validPartitionFormats, nil)
-
-	mp.sizeUnitDropDown = tview.NewDropDown().
-		SetLabel(uitext.DiskSizeUnitLabel).
-		SetOptions(validSizeUnits, nil)
-
-	// Restore the global style
-	tview.Styles = originalStyle
+	mp.sizeUnitInput, mp.sizeUnitInputReset = mp.enumInputBox(validSizeUnits)
+	mp.sizeUnitInput = mp.sizeUnitInput.SetLabel(uitext.DiskSizeUnitLabel).
```
Comment (PawelWMS) [inline toolkit/tools/imagegen/attendedinstaller/views/diskview/manualpartitionwidget/manualpartitionwidget.go]: Nit: could we follow the format from the deleted lines where each `Set*()` function was called on a separate line?

File: toolkit/tools/imagegen/attendedinstaller/views/diskview/manualpartitionwidget/manualpartitionwidget.go [hunk 3]
```diff
@@ -676,3 +660,42 @@ func (mp *ManualPartitionWidget) onNextButton() {
...
+
+	field.SetInputCapture(func(event *tcell.EventKey) *tcell.EventKey {
+		key := event.Key()
+		// Override movement left/right. Leave navigation keys intact. Consume all other input.
+		switch key {
+		case tcell.KeyLeft:
+			if index == 0 {
+				index = len(elements)
+			}
+			index--
```
Comment (PawelWMS) [inline toolkit/tools/imagegen/attendedinstaller/views/diskview/manualpartitionwidget/manualpartitionwidget.go:674]: Suggestion: this and `tcell.KeyRight` look like good places for a modulo operation `index = (elementsCount + index + [change]) % elementsCount`.

**Review by jslobodzian — APPROVED** [review]
:shipit:

**Review by jslobodzian — APPROVED** [review]

File: toolkit/tools/imagegen/attendedinstaller/views/diskview/manualpartitionwidget/manualpartitionwidget.go [hunk 4]
```diff
@@ -676,3 +660,42 @@ func (mp *ManualPartitionWidget) onNextButton() {
...
+// enumInputBox returns an input box that only allows values from elements to appear.
+// As well as resetter to reset the selection
+func (mp *ManualPartitionWidget) enumInputBox(elements []string) (field *tview.InputField, resetter func()) {
+	field = tview.NewInputField()
+	index := 0
+	// Initialize text with the first element
+	field.SetText(elements[index])
+
+	field.SetInputCapture(func(event *tcell.EventKey) *tcell.EventKey {
+		key := event.Key()
```
Comment (PawelWMS) [inline toolkit/tools/imagegen/attendedinstaller/views/diskview/manualpartitionwidget/manualpartitionwidget.go:667]: Nit: looks like you could pass `event.Key()` directly to `switch`.

File: toolkit/tools/imagegen/attendedinstaller/views/diskview/manualpartitionwidget/manualpartitionwidget.go [hunk 5]
```diff
@@ -676,3 +660,42 @@ func (mp *ManualPartitionWidget) onNextButton() {
...
+			return event
+		default:
+			mp.formNavBar.SetUserFeedback(uitext.EnumNavigationFeedback, tview.Styles.TertiaryTextColor)
+			return nil
+		}
+		field.SetText(elements[index])
+		return nil
+	})
+	return field, func() {
+		index = 0
```
Comment (PawelWMS) [inline toolkit/tools/imagegen/attendedinstaller/views/diskview/manualpartitionwidget/manualpartitionwidget.go]: Shouldn't we also do `field.SetText(elements[index])` here?

File: toolkit/tools/imagegen/attendedinstaller/views/diskview/manualpartitionwidget/manualpartitionwidget.go [hunk 6]
```diff
@@ -676,3 +655,39 @@ func (mp *ManualPartitionWidget) onNextButton() {
 		mp.nextPage()
 	}
 }
+
+// enumInputBox returns an input box that only allows values from elements to appear.
+func (mp *ManualPartitionWidget) enumInputBox(elements []string) *tview.InputField {
```
Comment (PawelWMS) [inline toolkit/tools/imagegen/attendedinstaller/views/diskview/manualpartitionwidget/manualpartitionwidget.go:660]: Neat!

**Pull request merged**